content/en/blog/2023/ambient-performance-savings/index.md

```diff
@@ -0,0 +1,135 @@
+- Ambient with L4 ztunnel and L7 waypoint proxies
+
+Spoilers! There are significant savings across the board for all ambient scenarios when compared to sidecars – up to 99% savings in usage and 90% in allocation.
+
+{{< image width="100%"
+    link="savings.png"
+    caption="Total Memory and CPU: Consumption and Allocation"
+    >}}
+
+Looking at total CPU and memory consumption, we have to remember that in the [ambient architecture](/blog/2022/introducing-ambient-mesh/), there are not sidecar containers for every application pod in the mesh. The result is that memory usage of Istio’s dataplane in the ztunnel-only ambient scenario uses *1%* of what is used in sidecar scenarios, and still only 10% when waypoints are added. Looking at CPU, ztunnel once again uses *1%* of what the sidecar scenario requires, and 15% when waypoints are deployed.
+
+Moving on to allocation, every sidecar resource has a default request of 100 millicores vCPU and 128 MB memory.  Assuming ztunnels and waypoint proxies have similar requests and limits as sidecars, that’s a *90% reduction* in allocated resources between L4 ambient and sidecar and 80% with waypoints!
+
+## Where the Savings Are Coming From
+
+Ambient mesh was designed to minimize resource requirements for users in their Kubernetes clusters. To explain how ambient does this, we must first clarify allocation versus utilization. When deploying a Kubernetes cluster on a hosted environment, nodes determine the overall capacity of the cluster and customer deployments, and pods are an allocation of that capacity. Utilization is a measure of how well this is done. As an architecture, sidecars interfere with effective utilization as they:
+
+- Define a high minimum for allocation at any scale
+- Can strand capacity by reserving more than is needed
+- Shift operational burden to adjust allocation if more than the default is needed
+
+The sidecar architecture forces this mode of allocation on users. Ambient solves these issues by leveraging a new architecture that separates the responsibilities of zero-trust networking and L7 policy handling. This is done with two new components to Istio: ztunnels and waypoint proxies.
+
+- [Ztunnels](/blog/2023/rust-based-ztunnel/) are a brand new Istio component written in Rust that are designed to be fast, secure, and lightweight. Ztunnels are deployed per node on a cluster and enable the most basic service mesh configurations for L4 features such as mTLS, telemetry, authentication, and L4 authorizations.
+- [Waypoint proxies](/blog/2023/waypoint-proxy-made-simple/) provide L7 mesh features such as VirtualService routing, L7 telemetry, and L7 authorizations policies. Waypoints are still based on Envoy and are deployed at the namespace level per ServiceAccount.
+
+These ztunnels and waypoint proxies work in tandem to replace sidecars in the Istio service mesh. So let’s take a closer look at how the two architectures compare in the tests above.
+
+## A Closer Look
```

Comment:
One thing that might be useful is to have a cost of "application" vs "application + ambient".

As an Istio adopter, I want to see "Adopting Istio will increase my bill by x%".

Granted that may be a pretty user-specific number... but may be useful to ballpark. If micro-services demo has reasonable `requests` it can be an ok approximation.

Reply:
So I do indirectly include that comparison in the `total-cpu` and `total-memory` charts.  Where yellow includes only istio data path pods (sidecars/ztunnel/waypoints) and green includes datapath pods + application pods.  The green charts are not really analyzed since it is application/user-specific